### PR TITLE
Fixes a problem with LayoutItem where dimensions are not always stored

### PR DESCRIPTION
**Pull request by johnspackman**
In https://github.com/qooxdoo/qooxdoo/issues/9553, the bug reported was that the container does not appear with non-zero sizes until an arbitrary event (eg browser window resize).

The size *was* given to a layout item at one point, but that the layout item choses to queue itself for resize so that the "height for width" calculation can be done.  The problem is that when that deferred resize happens, the item's container asks the item for it's current size - but because it has been deferred, the original sizes were not stored.

This fix just makes sure that the sizes are stored, and then continues with the deferred resize as before.

### CODE REVIEW

File: framework/source/class/qx/ui/core/LayoutItem.js [hunk 1]
```diff
@@ -516,6 +496,29 @@ qx.Class.define("qx.ui.core.LayoutItem",
...
+       * 
+       * Note that it is important that this happens after the above first pass at calculating a 
+       * computed size because otherwise getBounds() will return null, and this will cause an
+       * issue where the existing size is expected to have already been applied by the layout.
+       * See https://github.com/qooxdoo/qooxdoo/issues/9553  
+       */
+      if (this.getHeight() == null && this._hasHeightForWidth()) {
+        var flowHeight = this._getHeightForWidth(width);
+        
+        if (flowHeight != null && flowHeight !== this.__computedHeightForWidth)
```
Comment (derrell) [inline framework/source/class/qx/ui/core/LayoutItem.js]: `!=` with null... Are you checking for falsy, or do you mean to check explicitly for null?

File: uitests/issue9553/readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# issue9553
+
+undefined
+
+undefined
```
Comment (derrell) [inline uitests/issue9553/readme.md]: Is this file supposed to be committed as it is?

**Review by derrell — CHANGES_REQUESTED** [review]
see prior comments

File: framework/source/class/qx/ui/core/LayoutItem.js (hunk 1 above)
Comment (johnspackman) [reply framework/source/class/qx/ui/core/LayoutItem.js]: That code is just moved down - I agree that the comparisons are suspect, but it dates back to before 2011, so I've just kept it as it is

File: uitests/issue9553/readme.md (hunk 2 above)
Comment (johnspackman) [reply uitests/issue9553/readme.md]: oops, thats a bug in the skeleton generation

**Comment by oetiker** [comment]
do you need the pngs in the unittest ?

**Review by oetiker — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Pull request merged**